### PR TITLE
Fixes Neutralizing Gas / Intim interaction freezing the game

### LINKED ISSUE #4627: Game crashes when Neutralizing Gas mon is switched out vs Intimidate mon
### Description

Switching out a Pokémon with the ability Neutralizing Gas against a Pokémon with Intimidate crashes the game. This is repeatable.

The correct behavior should be for Intimidate's ability popout to occur before Weezing swaps, which is what is happening. The first breach from vanilla behavior is when Incineroar switches out. Here's Bulbapedia's description of the vanilla behavior:

> Once Neutralizing Gas is no longer applying, Abilities that activate when gained (i.e. [Abilities that activate when a Pokémon is switched in](https://bulbapedia.bulbagarden.net/wiki/Category:Abilities_that_activate_upon_entering_battle), such as [Intimidate](https://bulbapedia.bulbagarden.net/wiki/Intimidate_(Ability))) immediately pop up an activation message regardless of whether they have already done so, with the exceptions of [Unnerve](https://bulbapedia.bulbagarden.net/wiki/Unnerve_(Ability)) (which still applies but without displaying a message) and [Imposter](https://bulbapedia.bulbagarden.net/wiki/Imposter_(Ability)) (which only activates when switched in, not when gained). If Neutralizing Gas stops applying because the Pokémon with Neutralizing Gas is about to switch out, those Abilities activate before the switch occurs (e.g. if Intimidate is activated this way, it will decrease the Attack of the Pokémon with Neutralizing Gas before it switches out, rather than decreasing the Attack of the Pokémon who switches in to replace it).

mGBA_ZrC8ToBiKb

### Version

1.8.3 (Latest release)

### Upcoming/master Version

_No response_

### Discord contact info

Skolgrahd#6928

### PR DESCRIPTION
**Pull request by AlexOn1ine**
Fixes #4627

like #4668 but targeted to master. I've made new various since it is a temporary solution and didn't want to spend to much time on it.

### CODE REVIEW

File: test/battle/ability/intimidate.c [hunk 1]
```diff
@@ -210,3 +210,21 @@ SINGLE_BATTLE_TEST("Intimidate can not further lower opponents Atk stat if it is
         EXPECT_EQ(player->statStages[STAT_ATK], MIN_STAT_STAGE);
     }
 }
+
+SINGLE_BATTLE_TEST("[1]Intimidate activates when it's no longer effected by Neutralizing Gas")
```
Comment (cafei-uh) [inline test/battle/ability/intimidate.c:214]: *affected

Comment (AlexOn1ine) [inline test/battle/ability/intimidate.c:214]:
```suggestion
SINGLE_BATTLE_TEST("Intimidate activates when it's no longer affected by Neutralizing Gas")
```

Comment (AlexOn1ine) [reply test/battle/ability/intimidate.c]: thx

**Comment by ghoulslash** [comment]
Did 4061 fix the issue too? Should that just get pred to master?

**Comment by AlexOn1ine** [comment]
> Did 4061 fix the issue too? Should that just get pred to master?
Yeah, it did and would probably be better yeah.

**Comment by AlexOn1ine** [comment]
To avoid merge conflicts I'll close this. It is fixed in upcoming and should arrive soon on master.

**Pull request closed without merging**